<launch>

	<!-- Arguments -->
	<arg name="bagfile" default="bagfile.bag"/> <!-- Your bagfile here -->
	<arg name="camera" default="/stereo" /> <!-- The namespace where images are published -->
	<arg name="disparity_params" default="disparity_params.yaml"/> <!-- Camera calibration parameters -->

	<param name="/use_sim_time" value="true"/>

	<!-- Run the rosbag -->
	<node pkg="rosbag" type="play" name="rosbag" args="--clock $(arg bagfile)"/>

	<!-- Run the ROS package stereo_image_proc -->
	<group ns="$(arg camera)" >
		<node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc">
	    	<rosparam file="$(arg disparity_params)"/>
		</node>
	</group>

	<!-- Run the viso2_ros package -->
	<node pkg="viso2_ros" type="stereo_odometer" name="stereo_odometer" output="screen">
		<remap from="stereo" to="$(arg camera)"/>
		<remap from="image" to="image_rect"/>

		<!-- Matcher params -->
		<param name="nms_n" value="3" />
		<param name="nms_tau" value="50" />
		<param name="match_binsize" value="50" />
		<param name="match_radius" value="200" />
		<param name="match_disp_tolerance" value="2" />
		<param name="outlier_disp_tolerance" value="5" />
		<param name="outlier_flow_tolerance" value="5" />
		<param name="multi_stage" value="1" />
		<param name="half_resolution" value="1" />
		<param name="refinement" value="1" />

		<!-- Bucketing params -->
		<param name="max_features" value="2" />
		<param name="bucket_width" value="50" />
		<param name="bucket_height" value="50" />

		<!-- Stereo params -->
		<param name="ransac_iters" value="200" />
		<param name="inlier_threshold" value="1.5" />
		<param name="reweighting" value="1" />

		<!-- Reference frame -->
		<param name="ref_frame_change_method" value="0" />
		<param name="ref_frame_motion_threshold" value="5" />
		<param name="ref_frame_inlier_threshold" value="150" />
	</node>
	
</launch>
